Add Fireblocks integration guide to v2 docs (#352)
OpenZeppelin/defender-docs: docs/modules/ROOT/pages/guide/fireblock-defender-integration.adoc

= Fireblocks integration within Defender

> You can also submit transactions to Fireblocks from Defender directly. Fireblocks is an asset management solution that leverages multi-party computation for securing all treasury operations. 

Reviewer: ```suggestion
You can directly submit transactions to Fireblocks from Defender. Fireblocks is a robust asset management solution that utilizes multi-party computation to secure all treasury operations, ensuring enhanced security and efficiency.
```

[[pre-requisites]]
== Pre-requisites

* If you want leverage Fireblocks within Defender you can contact the OZ team to enable to Fireblocks integration for your account.

[[generate-csr-file]]
== 1. Generate CSR file

> . To utilise the feature you can navigate to the *Manage* page subsequently click *Approval Process* in the sidebar. If the Fireblocks integration is enabled for your account navigate to the *Integrations* tab, visible next to the All approval process.

Reviewer: ```suggestion
. To use this feature, navigate to the *Manage* page and click on *Approval Process* in the sidebar. If the Fireblocks integration is enabled for your account, go to the *Integrations* tab, which is located next to the *All Approval Process* tab.
```

+
image::guide-fireblocks-integration-tab.png[Integration tab]
+

> . Click *Generate new API Key for Fireblocks*, here you will need to generate a Certificate Signing Request (CSR) that we can use within Fireblocks file to utilise this feature and create API keys.

Reviewer: ```suggestion
. Click on *Generate new API Key for Fireblocks*. Here, you will need to generate a Certificate Signing Request (CSR), which will be used within the Fireblocks platform to enable this feature and create API keys.
```

+
image::guide-fireblocks-csr-modal.png[CSR Generation Modal]
This will trigger Defender to generate a public/private key-pair. The CSR is then generated and signed with the private key and securely stored to prevent leakage.

[[create-fireblocks-api-user]]
== 2. Create Fireblocks API user

> . First, you will need to import the CSR within the Fireblocks UI when creating a new API user. Note that the API user will require any role that can _at least initiate transactions, e.g. Signer.

Reviewer: ```suggestion
. First, you will need to import the CSR within the Fireblocks UI when creating a new API user. Note that the API user will require any role that can _at least_ initiate transactions, e.g. Signer.
```

+
image::guide-fireblocks-api-key.png[API Key generated]
+

[[setup-defender-integration]]
== 3. Connect Fireblocks with Defender
. First, navigate to the *Manage* page subsequently click *Approval Process* in the sidebar, the navigate to the *Integrations* tab. Over here click on the *Paste API Key from Fireblocks*.
+
image::guide-fireblock-paste-api-key.png[Insert API Key Defender]
+
. Insert the Fireblocks API key.
+
image::guide-fireblocks-edit-api-key.png[Insert API key, width=600]
+
NOTE: To submit a transaction to Fireblocks via Defender, ensure the correct permissions are set in Fireblocks, such as the relevant whitelisted addresses and the Transaction Access Policy (TAP). For example, you might need to whitelist the contract address you wish to interact with, as well as ensure that the newly created API user is allowed to interact with the relevant account and vaults (defined in the TAP).

[[create-approval-process]]
== 4. Create Approval Process

> . Once configured, you will be able to submit a transaction via a proposal. Select Fireblocks as the execution strategy, the API key and the vault you wish to initiate the transaction from. Once submitted, Defender will track the status of the transaction. Note, Defender will not allow you to approve or reject a transaction from the UI. This will require the Fireblocks mobile app, or console.

Reviewer: ```suggestion
. Once configured, you will be able to submit a transaction via a proposal. Select a Fireblocks approval process, the API key and the vault you wish to initiate the transaction from. Once submitted, Defender will track the status of the transaction. Note, Defender will not allow you to approve or reject a transaction from the UI. This is only possible via the Fireblocks mobile app or console.
```
